### ISSUE TITLE
How dataset label used for prediction in Alpha360 is calculated?

### ISSUE THREAD

**Issue by igor17400** [microsoft/qlib#1024]
## ❓ Questions and Help

I’m executing XGBoost and in order to better understand the data used for training in such model, which is provided by the DataHandler module, I downloaded it into .csv format and opened with pandas. However, I have been facing some data interpretation problems.

The workflow for this model has a column named `label` which, if I understood correctly, is provided by the variable `label` defined by the user as shown in the [link](https://github.com/microsoft/qlib/blob/5200ff520aeff6079faee7a0fa24a9217c08d806/examples/benchmarks/XGBoost/workflow_config_xgboost_Alpha360.yaml#L18).  In other words, this label variable is defined as `["Ref($close, -2) / Ref($close, -1) - 1”]`.

However, when I use qlib’s Data Retrieval to calculate such formula "Ref($close, -2) / Ref($close, -1) - 1” it returns a different value. 

_**Why is that?**_

_More info:_

<hr/>

Code used for qlib’s data retrieval:

```
df_ = D.features(D.instruments('csi300'), ['Ref($close, -2)/Ref($close, -1)-1'])
df_.loc[(['SH600000'], '2008-01-02'), :]
```

Value returned:

```
Date: 2008-01-02
Instrument: SH600000
Ref($close, -2) / Ref($close, -1) - 1: 0.042068
```

![Screen Shot 2022-03-28 at 21 42 47](https://user-images.githubusercontent.com/42355184/160510059-7ba26629-6661-42ca-a472-54e9772ebed0.png)

![Screen Shot 2022-03-28 at 22 22 06](https://user-images.githubusercontent.com/42355184/160513557-947bd095-08c8-466f-bead-6a0efaddffd3.png)

 <hr/>

Dataframe used for training in XGBoost:

```
Date: 2008-01-02
Symbol: SH600000
Ref($close, -2) / Ref($close, -1) - 1: 1.3606049822064057
```
<img width="1172" alt="Screen Shot 2022-03-28 at 21 38 48" src="https://user-images.githubusercontent.com/42355184/160509750-c5278ec8-8a79-42c0-a3bd-735030e1b2c1.png">

**Comment by al33m501** [comment]
I have the same question. I can't realize how LABEL0 is calculated

**Comment by hensejiang** [comment]
Yep, that was the question botherd me quite a while. Seems that Qlib is designed to generate predictions for BETTER performance in market rather than CLOSER to your target (aka label). And this makes sense to me when acknowledging the only strategy, Topkdropout  --- it cares only about the order of the instruments but not the pred scores.

**Comment by igor17400** [comment]
@al33m501 and @hensejiang I believe I have understood how LABEL0 is calculated using Aplha360. If you want read
the following [gist](https://gist.github.com/igor17400/bb97cb6ff34ab7706335ae32b2c9a053) or the PDF attached on this comment. 

Maybe we could open a PR to update the docs for a more detailed explanation such as the one I've made.

I'm not 100% sure if my explanation is correct, if you have any question or feedbacks feel free to make them!
I hope it helps :) 

[csrank_study.pdf](https://github.com/microsoft/qlib/files/8413261/csrank_study.pdf)

**Comment by al33m501** [comment]
@igor17400 Thank you for detailed explanation. If I understood everything correctly, we can't reverse normalize LABEL0 and it would be incorrect to use them as labels in  [model_performance_graph](https://qlib.readthedocs.io/en/latest/component/report.html#qlib.contrib.report.analysis_model.analysis_model_performance.model_performance_graph) for example (perhaps only I made such a mistake).

**Comment by igor17400** [comment]
@al33m501 can you provide a more detailed explanation for what you've said in the quote below?

> it would be incorrect to use them as labels in [model_performance_graph](https://qlib.readthedocs.io/en/latest/component/report.html#qlib.contrib.report.analysis_model.analysis_model_performance.model_performance_graph) for example

I couldn't clearly understand what you meant ;-;

**Comment by al33m501** [comment]
@igor17400 there is a module in qlib to evaluate model without building strategy - [model_performance_graph](https://qlib.readthedocs.io/en/latest/component/report.html#qlib.contrib.report.analysis_model.analysis_model_performance.model_performance_graph). We need to pass dataframe with predictions and **labels**. Until I came across this question, I was passing there LABEL0, which is **incorrect**.

**Comment by igor17400** [comment]
@al33m501 now I got it! 
I haven't executed this `model_performance_graph` module before. I've read the documentation you provided but I couldn't understand _how qlib can evaluate a model without building strategy._ Did you comprehend that?


If I understood correctly, it seems that the module gets the label provided by us in the `data_handler_config` under the `label` configuration.  So, for example, if we have configured the `data_handler_config` as shown below it would get the label as `Ref($close, -2)/Ref($close, -1) - 1)`.

```
data_handler_config: &data_handler_config
    start_time: 2008-01-01
    end_time: 2020-08-01
    fit_start_time: 2008-01-01
    fit_end_time: 2014-12-31
    instruments: *market
    infer_processors: []
    learn_processors:
        - class: DropnaLabel
        - class: CSRankNorm
          kwargs:
              fields_group: label
    label: ["Ref($close, -2)/Ref($close, -1) - 1)"]
```

Let's suppose our test period is `[2021-01-01, 2022-01-01]`.

The module `model_performance_graph` compares the results from the `Ref($close, -2)/Ref($close, -1) - 1)` computed for our test period with the results obtained form the model (ex: XGBoost, CatBoost,...) used to predict the label (score). 

That's why our dataframe passed as an argument into `model_performance_graph` has two columns `label` and `score` as shown below. 
- Score being the predict value and;
- Label  the actual calculated value from our test period

![Screen Shot 2022-04-09 at 13 49 33](https://user-images.githubusercontent.com/42355184/162583531-dfaac4d5-0cfd-4aa1-9d5d-b381174f1249.png)

Does that makes sense?
